src/main/java/com/CezaryZal/manager/filters/comparator/PasswordComparator.java

```diff
@@ -0,0 +1,14 @@
+package com.CezaryZal.manager.filters.comparator;
+
+import com.CezaryZal.exceptions.IncorrectLoginOrPasswordSecurityException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class PasswordComparator {
+
+    public void throwIsNotEqualsPassword(String passwordFromUserLogin, String userPasswordFromDb) {
+        if (!passwordFromUserLogin.equals(userPasswordFromDb)){
+            throw new IncorrectLoginOrPasswordSecurityException("Login or password incorrectly entered");
+        }
+    }
+}
```

Comment:
Ta klasa raczej nie powinna istnieć, masz w springsecurity mechanizm do tego.

public AuthenticationResponse authenticate(AuthenticationRequest authenticationRequest) {
        UsernamePasswordAuthenticationToken authenticationToken =
                new UsernamePasswordAuthenticationToken(authenticationRequest.getUsername(), authenticationRequest.getPassword());
        Authentication authentication = authenticationManager.authenticate(authenticationToken);
        if (authentication.isAuthenticated()) {
            String jwt = jwtBuilder.build(authentication, AuthenticationConstants.JWT_EXPIRE_IN_MINUTES);
            RefreshToken refreshToken = refreshTokenService.buildRefreshToken(authentication, AuthenticationConstants.REFRESH_TOKEN_EXPIRE_IN_MINUTES);
            refreshTokenService.addToken(refreshToken);
            return new AuthenticationResponse(jwt, refreshToken.getValue());
        }
        throw new LoginException("Invalid username or password");
    }




@Service
public class UserAuthenticationService implements AuthenticationManager {

    private final UserService userService;

    @Autowired
    public UserAuthenticationService(UserService userService) {
        this.userService = userService;
    }

    @Override
    public Authentication authenticate(Authentication authentication) throws AuthenticationException {
        Optional<User> userOptional = userService.getUserByUserName(authentication.getName());
        if (userOptional.isPresent()) {
            User user = userOptional.get();
            if (user.getPassword().equals(authentication.getCredentials())) {
                return new UsernamePasswordAuthenticationToken(authentication.getPrincipal(), null, user.getAuthorities());
            }
        }
        return new UsernamePasswordAuthenticationToken(authentication.getPrincipal(), authentication.getCredentials());
    }


Dodatkowo czy Twoje hasłą nie są przechowywane w bazie danych jako plain tekst?

Reply:
Ok, dzięki za pokazanie mechanizmu.
Niestety logikę i baze danych mam opartą na własnym obiekcie i podpięcie tego zajełoby trochę czasy dlatego narazie zostawie obecne rozwiazanie.
Napewno w przyszłości będę chciał ją wykorzystać